main.cpp

```diff
@@ -124,35 +160,43 @@ int main(void) {
     }
 
     // Creating resources, which can be written or read from the cloud
-    button_res = client.create_resource("3200/0/5501", "button_count");
-    button_res->set_value(0);
-    button_res->methods(M2MMethod::GET);
-    button_res->observable(true);
-    button_res->attach_notification_callback(button_callback);
-
-    pattern_res = client.create_resource("3201/0/5853", "blink_pattern");
-    pattern_res->set_value("500:500:500:500:500:500:500:500");
-    pattern_res->methods(M2MMethod::GET | M2MMethod::PUT);
-    pattern_res->attach_put_callback(pattern_updated);
+    res_button = client.create_resource("3200/0/5501", "button_count");
+    res_button->set_value(0);
+    res_button->methods(M2MMethod::GET);
+    res_button->observable(true);
+    res_button->attach_notification_callback(button_callback);
 
-    MbedCloudClientResource *blink_res = client.create_resource("3201/0/5850", "blink_action");
-    blink_res->methods(M2MMethod::POST);
-    blink_res->attach_post_callback(blink_callback);
+    res_led = client.create_resource("3201/0/5853", "led_state");
+    res_led->set_value(1);
+    res_led->methods(M2MMethod::GET | M2MMethod::PUT | M2MMethod::POST);
+    res_led->attach_put_callback(led_put_callback);
 
-    printf("Initialized Pelion Client. Registering...\n");
+    printf("Initialized Pelion Device Management Client. Registering...\n");
 
     // Callback that fires when registering is complete
     client.on_registered(&registered);
 
-    // Register with Pelion Device Management
+    // Register with Pelion DM
     client.register_and_connect();
 
-    // Placeholder for callback to update local resource when GET comes.
+    int i = 600; // wait up 60 seconds before attaching sensors and button events
+    while (i-- > 0 && !client.is_client_registered()) {
+        wait_ms(100);
+    }
+
+#ifdef USE_BUTTON
```

Comment:
BUTTON1 is always defined in the HAL these days, so this `USE_BUTTON` macro should not be needed.